### DISCUSSION TITLE
How can I create a zip file but starting from particular subdirectory?

### DISCUSSION THREAD

**Discussion by vittoso** [montudor/action-zip#3] [Q&A] [answered]
Not really an issue, but maybe anyone can help me to figure out what's wrong...

I have a repo with some files and a subdirectory for an AWS lambda function.
Folder tree is 
\ - repo root
... - other folders
\lambda\awszone\index.js   
\lambda\awszone\node-modules\...
\lambda\awszone\....  

I'd like to create a zip with files from \lambda\awszone\ and its subfolders, but the zip should not contain the \lambda\awszone\ part of the path, that is: index.js should be on the root of the generated .zip file.
Something like:
index.js   
node-modules\...
....  

I've tried using your examples with the -r recursive option, also tried with -j (error for file with the same name) and -D (didn't change anything)... All my test create zip file containing the full path (\lambda\awszone\).

Can anybody please give me some suggestions?

Thank you very much!

**Comment by montudor** [comment]
I can't test it right now, however I believe if you change the `args` to
```bash
cd /lambda/awszone && zip -r output.zip *
```
then it should work. Please note, this means the destination of the zip will probably be in the `/lambda/awszone` if you specify a relative path as shown above, so you should probably use an absolute path instead of `output.zip`.

**Comment by vittoso** [comment]
It looks as "cd" command is not present in the docker instance.

The error is:

> /usr/bin/docker: Error response from daemon: OCI runtime create failed: container_linux.go:346: starting container process caused "exec: \"cd\": executable file not found in $PATH": unknown.

I tried args
` cd ./lambda/awszone/ && zip  -r ./lambdazip/lambda.zip  *`

**Comment by montudor** [comment]
I'm gonna have to work on adding this to the project. Perhaps for now you could (in a separate step) copy the files you need to the working directory and then zip them from there. Alternatively you could install zip and cd in using the run commands.

**Comment by ranganapeiris** [comment]
Hi Monte,
Can you please provide an update on this? I am also eagerly waiting for this fix. I have many files in my directory and just copying them to working directory then zipping it is quite difficult.

**Comment by montudor** [comment]
@ranganapeiris unfortunately I am pretty occupied at the moment but I will get working on a fix asap

**Comment by sanchezz985** [comment]
Hi everyone, is this resolved ?

**Comment by montudor** [comment]
An update is in the works and v2 should be ready soon. I apologise for the amount of time it's taking. This year has given me a limited amount of time to work on this project.

**Comment by virtualcodewarrior** [comment]
cd is actually a shell buildin and not a command and that is why it fails with cd
The following seems to work for me however:

`args: sh -c "cd doNotIncludeThisFolder && zip -qq -r ../output.zip ."`

**Comment by crqrdotcom** [comment]
> cd is actually a shell buildin and not a command and that is why it fails with cd
> The following seems to work for me however:
> 
> `args: sh -c "cd doNotIncludeThisFolder && zip -qq -r ../output.zip ."`

Confirmed this "workaround" and works perfectly. Thanks!

**Comment by sdemjanenko** [comment, marked as answer]
You can also use multiple steps in the Github job workflow. Layers from the previous step are available in later steps.
```
    - name: Install zip
      uses: montudor/action-zip@v0.1.0

    - name: Zip output
      run: zip -qq -r function.zip dist node_modules package.json
      working-directory: backend/some-module
```

**Comment by montudor** [comment]
This seems like a good clean solution to me. For now, I would say this is the way to go.

I have added this example to the README. Thanks @sdemjanenko!
